### PR TITLE
[#4718] fix(iceberg):  use unified logic to transform catalog backend name to handle the renaming of catalog

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?

1. Spark,Trino, Iceberg catalog and Iceberg REST server use  `getCatalogBackendName` to get catalog backend name.
2. change the default backend name to catalog backend, like `jdbc`, it will not change after rename.

### Why are the changes needed?

Fix: #4718 

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
1. create a jdbc catalog with catalog backend name, check whether can see the schema after rename
1. create a jdbc catalog without catalog backend name, check whether can see the schema after rename

### CODE REVIEW

File: docs/lakehouse-iceberg-catalog.md [hunk 1]
```diff
@@ -36,12 +36,12 @@ Builds with Apache Iceberg `1.5.2`. The Apache Iceberg table format version is `
...
+| `warehouse`                                        | Warehouse directory of catalog. `file:///user/hive/warehouse-hive/` for local fs or `hdfs://namespace/hdfs/path` for HDFS.                                                                                                                       | (none)                            | Yes                                                         | 0.2.0         |
+| `catalog-backend-name`                             | The catalog name passed to underlying Iceberg catalog backend. Catalog name in JDBC backend is used to isolate namespace and tables.                                                                                                             | `catalog-backend` property value. | No                                                          | 0.5.2         |
```
Comment (diqiu50) [inline docs/lakehouse-iceberg-catalog.md:44]: The default value is "**catalog-backend**"  or  "**`catalog-backend` property value**".

File: catalogs/catalog-common/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergPropertiesUtils.java [hunk 2]
```diff
@@ -74,4 +76,22 @@ public static Map<String, String> toIcebergCatalogProperties(
...
+  public static String getCatalogBackendName(Map<String, String> catalogProperties) {
+    String backendName = catalogProperties.get(IcebergConstants.CATALOG_BACKEND_NAME);
+    if (backendName != null) {
+      return backendName;
+    }
+
+    String catalogBackend = catalogProperties.get(IcebergConstants.CATALOG_BACKEND);
+    return Optional.ofNullable(catalogBackend)
+        .map(s -> s.toLowerCase(Locale.ROOT))
+        .orElse("memory");
```
Comment (diqiu50) [inline catalogs/catalog-common/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergPropertiesUtils.java:95]: What's the meaning of the default value `memory"?

File: trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergConnectorAdapter.java [hunk 3]
```diff
@@ -46,7 +46,6 @@ public IcebergConnectorAdapter() {
   @Override
   public Map<String, String> buildInternalConnectorConfig(GravitinoCatalog catalog)
       throws Exception {
-    catalog.getProperties().put("catalog-name", catalog.getName());
```
Comment (diqiu50) [inline trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergConnectorAdapter.java:49]:
```java

/** Help Apache Gravitino connector access CatalogMetadata from Gravitino client. */
public class GravitinoCatalog {

  private static ObjectMapper objectMapper = new ObjectMapper();

  private final String metalake;
  private final String provider;
  private final String name;
  private final Map<String, String> properties;
  private final long lastModifiedTime;

  static {
    objectMapper =
        JsonMapper.builder()
            .disable(MapperFeature.AUTO_DETECT_CREATORS)
            .disable(MapperFeature.AUTO_DETECT_FIELDS)
            .disable(MapperFeature.AUTO_DETECT_SETTERS)
            .disable(MapperFeature.AUTO_DETECT_GETTERS)
            .build();
  }

  public GravitinoCatalog(String metalake, Catalog catalog) {
    this.metalake = metalake;
    this.provider = catalog.provider();
    this.name = catalog.name();
    this.properties = new HashMap<>(catalog.properties());
    Instant time =
        catalog.auditInfo().lastModifiedTime() == null
            ? catalog.auditInfo().createTime()
            : catalog.auditInfo().lastModifiedTime();
    lastModifiedTime = time.toEpochMilli();
  }
```
After we remove this code,  The  properties map should not copy from the original catalog properties.

File: docs/lakehouse-iceberg-catalog.md (hunk 1 above)
Comment (FANNG1) [reply docs/lakehouse-iceberg-catalog.md]: The property value of `catalog-backend`, and add an example

File: catalogs/catalog-common/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergPropertiesUtils.java (hunk 2 above)
Comment (FANNG1) [reply catalogs/catalog-common/src/main/java/org/apache/gravitino/catalog/lakehouse/iceberg/IcebergPropertiesUtils.java:95]: the default catalog backend is `memory`

File: trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergConnectorAdapter.java (hunk 3 above)
Comment (FANNG1) [reply trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergConnectorAdapter.java:49]: make changes like that?
```
this.properties  = catalog.properties()
```

Comment (FANNG1) [reply trino-connector/trino-connector/src/main/java/org/apache/gravitino/trino/connector/catalog/iceberg/IcebergConnectorAdapter.java:49]: updated

File: docs/lakehouse-iceberg-catalog.md (hunk 1 above)
Comment (diqiu50) [reply docs/lakehouse-iceberg-catalog.md]: Is it better to use only values like `catalog-backend`?

Comment (FANNG1) [reply docs/lakehouse-iceberg-catalog.md]: seems a little odd

File: docs/lakehouse-iceberg-catalog.md [hunk 4]
```diff
@@ -36,12 +36,12 @@ Builds with Apache Iceberg `1.5.2`. The Apache Iceberg table format version is `
...
+| `warehouse`                                        | Warehouse directory of catalog. `file:///user/hive/warehouse-hive/` for local fs or `hdfs://namespace/hdfs/path` for HDFS.                                                                                                                       | (none)                                                                         | Yes                                                         | 0.2.0         |
+| `catalog-backend-name`                             | The catalog name passed to underlying Iceberg catalog backend. Catalog name in JDBC backend is used to isolate namespace and tables.                                                                                                             | The property value of `catalog-backend`, like `jdbc` for JDBC catalog backend. | No                                                          | 0.5.2         |
```
Comment (diqiu50) [inline docs/lakehouse-iceberg-catalog.md:45]: Is the default catalog backend  `memory`

Comment (FANNG1) [reply docs/lakehouse-iceberg-catalog.md:45]: It depondes on `catalog-backend`, not `memory`

**Review by diqiu50 — APPROVED** [review]
LGTM

**Pull request merged**